public/app/features/alerting/unified/components/alert-groups/AlertDetails.tsx

```diff
@@ -16,34 +18,38 @@ export const AlertDetails: FC<AmNotificationsAlertDetailsProps> = ({ alert, aler
   return (
     <>
       <div className={styles.actionsRow}>
-        {alert.status.state === AlertState.Suppressed && (
-          <LinkButton
-            href={`${makeAMLink(
-              '/alerting/silences',
-              alertManagerSourceName
-            )}&silenceIds=${alert.status.silencedBy.join(',')}`}
-            className={styles.button}
-            icon={'bell'}
-            size={'sm'}
-          >
-            Manage silences
-          </LinkButton>
-        )}
-        {alert.status.state === AlertState.Active && (
-          <LinkButton
-            href={makeLabelBasedSilenceLink(alertManagerSourceName, alert.labels)}
-            className={styles.button}
-            icon={'bell-slash'}
-            size={'sm'}
-          >
-            Silence
-          </LinkButton>
-        )}
-        {alert.generatorURL && (
-          <LinkButton className={styles.button} href={alert.generatorURL} icon={'chart-line'} size={'sm'}>
-            See source
-          </LinkButton>
-        )}
+        <Authorize actions={[AccessControlAction.AlertingInstanceCreate, AccessControlAction.AlertingInstanceUpdate]}>
+          {alert.status.state === AlertState.Suppressed && (
+            <LinkButton
+              href={`${makeAMLink(
+                '/alerting/silences',
+                alertManagerSourceName
+              )}&silenceIds=${alert.status.silencedBy.join(',')}`}
+              className={styles.button}
+              icon={'bell'}
+              size={'sm'}
+            >
+              Manage silences
+            </LinkButton>
+          )}
+          {alert.status.state === AlertState.Active && (
+            <LinkButton
+              href={makeLabelBasedSilenceLink(alertManagerSourceName, alert.labels)}
+              className={styles.button}
+              icon={'bell-slash'}
+              size={'sm'}
+            >
+              Silence
+            </LinkButton>
+          )}
+        </Authorize>
+        <Authorize actions={[AccessControlAction.DataSourcesExplore]}>
```

Comment:
I think this one is incorrect; I believe the `generatorURL` points to the edit page of an alert so we should check for `AccessControlAction.AlertingInstanceUpdate`

Reply:
Yes that is the case for Grafana managed alerts, but for Cortex managed alerts it links to an explore link with the query. I'm thinking we'd need to check something along the lines of...

```ts
<Authorize
          actions={
            isExternalAM
              ? [AccessControlAction.DataSourcesExplore]
              : [AccessControlAction.AlertingInstanceUpdate]
          }
        >
...
```

what do you think @gillesdemey ?

Reply:
Oh – I had no idea those were different for Cortex (and I assume Loki) alerts 🙈 I think your suggestion is pretty much spot-on Nathan 👍

Reply:
I wonder if it should be `alert.instances.external:write` instead?